docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,554 @@
+    "> Note: To ensure that the latest changes are included, make sure you save your notebook before creating/updating the function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "26f3d35b",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "notebook_fn = single_source_project.set_function(\n",
+    "    name=\"notebook\", \n",
+    "    kind=\"job\", \n",
+    "    image=\"mlrun/mlrun\", \n",
+    "    handler=\"handler\"\n",
+    ")\n",
+    "\n",
+    "run = single_source_project.run_function(notebook_fn)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84cf9258",
+   "metadata": {},
+   "source": [
+    "<a id=\"multiple-source-files\"></a>\n",
+    "### Multiple Source Files (Git/Archive)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "773d7bef",
+   "metadata": {},
+   "source": [
+    "If your code requires additional files or external libraries, you will need to use a source that supports multiple files such as Git, an archive (zip/tar/etc.), or V3IO file share. This approach (especially using a Git repo) pairs well with MLRun Projects.\n",
+    "\n",
+    "With this approach, you must provide `with_repo=True` into `set_function`. This instructs MLRun to load source code from the git repo/archive/file share associated with the project.\n",
+    "\n",
+    "There are two options to load the code:\n",
+    "1. **Static**: The function is built once. *This is the preferred approach for production workloads*\n",
+    "> Build the function once via `mlrun.build_function(fn)`\n",
+    "2. **Runtime**: The function will pull the source code at runtime. *This is is a simpler approach during development that allows for making code changes without re-building the image each time*\n",
+    "> Enable this feature via `project.set_source(source=None, pull_at_runtime=True)`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ac08010",
+   "metadata": {},
+   "source": [
+    "#### Git Repo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8c7290be",
+   "metadata": {},
+   "source": [
+    "The Git repo is provided when creating the project like so:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "21426cf4",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "git_project = mlrun.load_project(\n",
+    "    name=\"git-project\",\n",
+    "    context=\"./git_project\",\n",
+    "    url=\"git://github.com/mlrun/project-archive\",\n",
+    "    clone=True\n",
+    ")\n",
+    "\n",
+    "git_project.set_source(source=None, pull_at_runtime=True)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9f5999c",
+   "metadata": {},
+   "source": [
+    "Then, you can use `set_function` with `with_repo=True` create the function. In this case, there is a Python file called `job_func.py` in the root directory of the git repo:"
```

Comment:
Then, you can use `set_function` with `with_repo=True` to create the function. In this case, there is a Python file called `job_func.py` in the root directory of the git repo: